jni/src/commons.cpp

```diff
@@ -107,3 +114,140 @@ int knn_jni::commons::getIntegerMethodParameter(JNIEnv * env, knn_jni::JNIUtilIn
 
     return defaultValue;
 }
+
+// FP32 → FP16
+void knn_jni::commons::convertFP32ToFP16(knn_jni::JNIUtilInterface* jniUtil,
+                                         JNIEnv* env,
+                                         jfloatArray fp32Array,
+                                         jbyteArray fp16Array,
+                                         jint count) {
+    if (count <= 0) return;
+
+    jfloat* src_f32 = reinterpret_cast<jfloat*>(env->GetPrimitiveArrayCritical(fp32Array, nullptr));
+    jbyte* dst_bytes = reinterpret_cast<jbyte*>(env->GetPrimitiveArrayCritical(fp16Array, nullptr));
+    const float* src = reinterpret_cast<const float*>(src_f32);
+    uint16_t* dst = reinterpret_cast<uint16_t*>(dst_bytes);
+
+    int i = 0;
+
+#if defined(__aarch64__)
```

Comment:
Can we move each of these extension-specific implementations to a separate file? Like in this faiss code https://github.com/facebookresearch/faiss/blob/main/faiss/utils/distances_fused/distances_fused.cpp#L30-L39.